### ISSUE TITLE
Access permission query results from client

### ISSUE THREAD

**Issue by kbrandwijk** [graphcool/framework#231]
**TL;DR**: Expose the result of the permission queries to the client, by defining a new query for each Type, and by exposing a special field in queries, both returning effective permissions based on the current instance or Type.
---
Permission queries are a very powerful way to specify the permission rules for CRUD operations. However, they are only executed when the operation is performed. In a client app, I would like to re-use the logic defined in the permission queries to know beforehand if a user can perform a CRUD operation on an object.

Let's consider a simple app: A user can create Posts, and other users can comment on them.
Rules: 
1. A user can only edit his own posts, and only if the post has no comments yet.
2. A user can comment on all posts, except his own.
3. A user can only delete his own comments.
4. A user can not create a post on a topic he has no experience on.

All of these rules are easy to implement using permission queries.

In the client, certain buttons need to be enabled/disabled based on these rules. Right now, that means duplicating the logic for these rules in the client. 

Proposal:
- Expose functionality to run all permission queries for an intended CRUD operation, returning a boolean indicating if the permission is granted or not.

However, permission queries for create and update can be based on the input (the object that is to be created, or the new state of an object that is to be updated). This is only a problem for create permissions, because we don't have that new instance yet. In order for this approach to work, create permission queries need an additional flag indicating if they are instance-based or not. This allows all permission queries that are not instance-based to run beforehand.

So in short,
- Expose functionality to run all permission queries for an intended operation, always passing an object instance for 'update' and 'delete', optionally passing an object instance for 'create'
- Extend permission query definition for 'create' permission queries with a flag 'instance based', so when no instance is passed, only non-instance based permission queries will run

Bonus: figure out a way to include this information in the initial query (maybe using a similar approach to the _metadata funcionality, allowing the user to put the _permissions field anywhere in the query hierarchy to get the effective permissions on that object in the result tree).

**Comment by kbrandwijk** [comment]
Implemented by exposing the permissions endpoint.
